framework/include/csg/CSGSurface.h

```diff
@@ -0,0 +1,121 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "MooseApp.h"
+#include "MooseEnum.h"
+
+namespace CSG
+{
+
+/**
+ * CSGSurface creates an internal representation of a Constructive Solid Geometry (CSG)
+ * surface, represented as some polynomial in x, y, and z
+ */
+class CSGSurface
+{
+public:
+  /// Enum for the direction of the half-space being represented by a point and surface
+  enum class Direction
```

Comment:
Note, this was the only `enum class` that I did not change into a MooseEnum type. It didn't seem to make sense for this case because Direction is not an attribute on the surface to be set like the other MooseEnum types are. This one exists to restrict the options for direction to just positive or negative.